config/routes.rb

```diff
@@ -7,6 +7,11 @@
   root "users#index"
 
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :new, :create]
+  end
+
+  resources :posts do
+    resources :comments, only: [:new,:create]
+    resources :likes, only: [:create]
   end
 end
```

Comment:
- [ ] While I was migrating i.e run `rails db:migrate`, i got the below error 👇🏾 

![image](https://github.com/Tobby8629/Blog_app/assets/84156714/b239bb7c-a518-4819-a272-fee2b895a76e)

This simply means, that within the **_CreateComments_** migration file, you are referencing a `posts` that has not been created yet because the migration file that will create the **Post table** is after this **_CreateComments_** migration file. To solve this, you will have to delete your migration files and generate them in the proper order - User, Posts, Comments, Likes, ...etc

Reply:
Good day @bobb-Rob  thank you very much for the review, I have successfully fix the first correction and I really appreciate the correction. As regards to second correction I tried working on fixing it and I did not even get the error but a reviewer on my earlier had the same issue and said he was able to fix it using **rails db:setup** .  I am  suggesting if you can try working with the same command to see if it works
<img width="571" alt="rails-solution" src="https://github.com/Tobby8629/Blog_app/assets/108295038/96b5a574-c011-4352-a021-0090c31005ac">